pandas/_libs/tslibs/parsing.pyx

```diff
@@ -1210,3 +1216,25 @@ cpdef str get_rule_month(str source):
         return "DEC"
     else:
         return source.split("-")[1]
+
+cdef inline object _parse_today_now(str date_string):
+    """
+    Parse special case date inputs - "today", "now"
+    and if present return corresponding datetime object
+
+    Parameters
+    ----------
+    date_string : str
+
+    Returns:
+    --------
+    datetime or None
+    """
+    cdef:
+        object res = None
+
+    if date_string == "now":
```

Comment:
IMO I think this logic can directly go in `parse_datetime_string`

Also, this should probably return datetime objects, not Timestamp objects